encord_agents/core/utils.py

```diff
@@ -136,33 +142,48 @@ def download_asset(lr: LabelRowV2, frame: int | None) -> Generator[Path, None, N
         The file path for the requested asset.
 
     """
-    video_item, images_list = lr._project_client.get_data(lr.data_hash, get_signed_url=True)
-    if lr.data_type in [DataType.VIDEO, DataType.IMAGE] and video_item:
-        url = video_item["file_link"]
-    elif lr.data_type == DataType.IMG_GROUP and images_list:
+    url: str | None = None
+    if lr.data_link is not None and lr.data_link[:5] == "https":
+        url = lr.data_link
+    elif lr.backing_item_uuid is not None:
+        storage_item = get_user_client().get_storage_item(lr.backing_item_uuid, sign_url=True)
+        url = storage_item.get_signed_url()
+
+    # Fallback for native image groups (they don't have a url)
+    is_image_sequence = lr.data_type == DataType.IMG_GROUP
+    if url is None:
+        is_image_sequence = False
+        _, images_list = lr._project_client.get_data(lr.data_hash, get_signed_url=True)
+        if images_list is None:
+            raise ValueError("Image list should not be none for image groups.")
         if frame is None:
             raise NotImplementedError(
                 "Downloading entire image group is not supported. Please contact Encord at support@encord.com for help or submit a PR with an implementation."
             )
-        url = images_list[frame]["file_link"]
-    else:
-        raise ValueError(f"Couldn't load asset of type {lr.data_type}")
+        image = images_list[frame]
+        url = cast(str | None, image.file_link)
+
+    if url is None:
+        raise ValueError("Failed to get a signed url for the asset")
 
     response = requests.get(url)
     response.raise_for_status()
 
-    suffix = _guess_file_suffix(url, lr)
+    _, suffix = _guess_file_suffix(url, lr)
     file_path = Path(lr.data_hash).with_suffix(suffix)
```

Comment:
Maybe worth logging if there's something already sitting at that file

Reply:
Also do we wanna acquire a temporary directory that holds the folder rather than writing to some path within the agents library?
Feels safer somehow